crates/shared/src/oneinch_api.rs

```diff
@@ -37,6 +37,114 @@ impl<const MIN: usize, const MAX: usize> Display for Amount<MIN, MAX> {
     }
 }
 
+// The `Display` implementation for `H160` unfortunately does not print
+// the full address and instead uses ellipsis (e.g. "0xeeee…eeee"). This
+// helper just works around that.
+fn addr2str(addr: H160) -> String {
+    format!("{:?}", addr)
+}
+
+/// A query to get a quote forl a sell order with 1Inch.
+#[derive(Clone, Debug)]
+pub struct SellOrderQuoteQuery {
+    /// Contract address of a token to sell.
+    pub from_token_address: H160,
+    /// Contract address of a token to buy.
+    pub to_token_address: H160,
+    /// Amount of a token to sell, set in atoms.
+    pub amount: U256,
+    /// List of protocols to use for the quote. Default: all protocols
+    pub protocols: Option<Vec<String>>,
+    /// Percentage how much of the from_token_address amount should be sent to the referrer
+    /// address. Values: [0, 3], Default 0.0
+    pub fee: Option<f64>,
+    /// Maximum amount of gas for a swap. Default: 11500000
+    pub gas_limit: Option<Amount<0, 11500000>>,
+    /// Which tokens should be used for intermediate trading hops.
+    pub connector_tokens: Option<Vec<H160>>,
+    /// Maximum number of token-connectors to be used in a transaction. Default: 2
+    pub complexity_level: Option<Amount<0, 3>>,
+    /// Limit maximum number of main route parts. Default: 20
+    pub main_route_parts: Option<Amount<1, 50>>,
+    /// Limit the number of virtual split parts.
+    pub virtual_parts: Option<Amount<1, 500>>,
+    /// Limit maximum number of parts each main route part can be split into. Default: 20
+    pub parts: Option<Amount<1, 100>>,
+    /// Gas price in atoms. Default: "fast" from network
+    pub gas_price: Option<U256>,
```

Comment:
nit: What are atoms? 
Do you know when or how 1inch uses gas_price data?

Reply:
Atoms are smallest units of tokens. For example, `USDC` has 6 decimals. So a "base" amount of `1.50` corresponds to `1500000` atoms. Similarly, for `DAI` with 18 decimals, an amount of `42.1337` corresponds to `42133700000000000000` atoms.

Its equivalent to `wei` for `(W)ETH`.

Reply:
Atoms are the smallest divisible unit. I copied the term "atoms" from the `SwapQuery` comments but using the language of the 1Inch API makes more sense. 👍 
Now it's smallest divisible unit.
Not sure how 1Inch uses this data.